tests/rptest/tests/datalake/datalake_e2e_test.py

```diff
@@ -399,4 +399,7 @@ def test_lag_metrics(self, cloud_storage_type):
             dl.catalog_service.start()
 
             self.wait_for_lag(m, DatalakeMetricsTest.translation_lag, 0)
-            self.wait_for_lag(m, DatalakeMetricsTest.commit_lag, 0)
+
+            time.sleep(5)
+            dl.produce_to_topic(self.topic_name, 1, msg_count=count)
+            self.wait_for_lag(m, DatalakeMetricsTest.commit_lag, count)
```

Comment:
Could we end up committing so fast that we don't see the lag? Especially if we've reduced the commit interval?

Reply:
> could we end up committing so fast that we don't see the lag?

No, but for a weird reason. I put a note about this somewhere (commit message maybe), but based on my reading this seems to be a fundamental limitation of the new scheduling loop.

Basically if we `wait_for_data_to_translate` on an idle topic before the last batch of files are committed (and `mark_file_committed_update` applied), we won't see the updated `last_committed_offset` until we re-enter the translation loop.

So the reason we always see the lag here is that the lag sticks to `count` basically indefinitely (assuming we don't produce any more records). It would be better if this weren't the case, but I don't see an obvious solution. Any ideas?